cmd/host-agent/main.go

```diff
@@ -45,87 +48,75 @@ func (p *program) Stop(s service.Service) error {
-	if err != nil {
-		logger.Fatal("could not create OS service", zap.Error(err))
-	}
-
-	err = s.Run()
-	if err != nil {
-		logger.Error("error after running the service", zap.Error(err))
+	if err := p.collector.Run(context.Background()); err != nil {
+		p.logger.Error("collector server run finished with error", zap.Error(err))
 	}
 }
 
-func app(logger *zap.Logger) *cli.App {
-	var apiKey, target, configCheckInterval, apiURLForConfigCheck string
-	var hostTags string
-	var dockerEndpoint string
-	var enableSyntheticMonitoring bool
-	flags := []cli.Flag{
+func getFlags(cfg *agent.Config) []cli.Flag {
+	return []cli.Flag{
 		altsrc.NewStringFlag(&cli.StringFlag{
 			Name:        "api-key",
 			EnvVars:     []string{"MW_API_KEY"},
 			Usage:       "Middleware API key for your account.",
-			Destination: &apiKey,
+			Destination: &cfg.ApiKey,
 		}),
 		altsrc.NewStringFlag(&cli.StringFlag{
 			Name:        "target",
 			EnvVars:     []string{"MW_TARGET"},
 			Usage:       "Middleware target for your account.",
-			Destination: &target,
+			Destination: &cfg.Target,
 		}),
 		altsrc.NewBoolFlag(&cli.BoolFlag{
 			Name:        "enable-synthetic-monitoring",
 			EnvVars:     []string{"MW_ENABLE_SYNTHETIC_MONITORING"},
-			Destination: &enableSyntheticMonitoring,
+			Destination: &cfg.EnableSyntheticMonitoring,
 		}),
 		altsrc.NewStringFlag(&cli.StringFlag{
 			Name:    "config-check-interval",
 			EnvVars: []string{"MW_CONFIG_CHECK_INTERVAL"},
 			Usage: "Duration string to periodically check for configuration updates." +
 				"Setting the value to 0 disables this feature.",
-			Destination: &configCheckInterval,
+			Destination: &cfg.ConfigCheckInterval,
 			DefaultText: "60s",
 		}),
 		altsrc.NewStringFlag(&cli.StringFlag{
 			Name:        "docker-endpoint",
 			EnvVars:     []string{"MW_DOCKER_ENDPOINT"},
 			Usage:       "Set the endpoint for Docker socket if different from default",
-			Destination: &dockerEndpoint,
+			Destination: &cfg.DockerEndpoint,
 			DefaultText: "unix:///var/run/docker.sock",
 			Value:       "unix:///var/run/docker.sock",
 		}),
 		altsrc.NewStringFlag(&cli.StringFlag{
 			Name:        "api-url-for-config-check",
 			EnvVars:     []string{"MW_API_URL_FOR_CONFIG_CHECK"},
-			Destination: &apiURLForConfigCheck,
+			Destination: &cfg.ApiURLForConfigCheck,
 			DefaultText: "https://app.middleware.io",
 			Value:       "https://app.middleware.io",
 			Hidden:      true,
 		}),
 		altsrc.NewStringFlag(&cli.StringFlag{
 			Name:        "host-tags",
+			Usage:       "Tags for this host",
 			EnvVars:     []string{"MW_HOST_TAGS"},
-			Destination: &hostTags,
+			Destination: &cfg.HostTags,
+		}),
+		altsrc.NewStringFlag(&cli.StringFlag{
+			Name:        "logfile",
+			Usage:       "Log file to store Middleware agent logs",
+			EnvVars:     []string{"MW_LOGFILE"},
+			Destination: &cfg.Logfile,
+			DefaultText: "",
+			Value:       "",
+		}),
+
+		altsrc.NewIntFlag(&cli.IntFlag{
+			Name:        "logfile-size",
```

Comment:
This flag will allow us to rotate the logfile after the file has grown to this size.